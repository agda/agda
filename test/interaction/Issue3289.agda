{-# OPTIONS --postfix-projections #-}

open import Agda.Builtin.Nat

record R : Set where
  field
    x : Nat → Nat

open R {{...}}

f : R
f .x = {!!}

-- WAS: Case splitting on result produces the following nonsense:
-- f ⦃ .x ⦄ x₁ = ?

-- SHOULD: produce
-- f .x x₁ = ?
