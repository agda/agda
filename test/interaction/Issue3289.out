(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Nat → Nat " nil)
((last . 1) . (agda2-goals-action '(0)))
((last . 2) . (agda2-make-case-action '("f .x x₁ = ?")))
((last . 1) . (agda2-goals-action '(0)))
